core/commonMain/src/implementations/immutableMap/TrieNode.kt

```diff
@@ -603,8 +612,11 @@ internal class TrieNode<K, V>(
                 }
 
                 otherNode.hasNodeAt(positionMask) -> {
-                    val before = otherNode.nodeAtIndex(otherNode.nodeIndex(positionMask))
-                    before.mutablePutAllFromOtherNodeCell(this, positionMask, shift, intersectionCounter, mutator)
+                    val thisKeyIndex = this.entryKeyIndex(positionMask)
+                    val thisKey = this.keyAtIndex(thisKeyIndex)
+                    val thisValue = this.valueAtKeyIndex(thisKeyIndex)
+                    val before = makeNode(thisKey.hashCode(), thisKey, thisValue, shift, mutator.ownership)
```

Comment:
`TrieNode` does not allow nodes with a single entry (except the root node). Many functions rely on this invariant. Although `before` is a throwaway/temporary node, it signals that there is a problem in `mutablePutAllFromOtherNodeCell`.

https://github.com/Kotlin/kotlinx.collections.immutable/pull/116 is my attempt to fix the bug. Could you please review it and share your thoughts?

Reply:
I have closed this in favor of yours. 

I realized I was violating an invariant but it wasn't for long as the putAll... call would restore the invariant immediately but this requires an allocation which your code avoids so it is better in any case. Mine was just the smallest change that worked (or, at least, I think it worked...).